### DISCUSSION TITLE
`git filter-branch` alternative

### DISCUSSION THREAD

**Discussion by acomagu** [jj-vcs/jj#4757] [Ideas]
How can I execute batch operations like rename file, npm install or executing any other shell script for whole branch in jj?

I want `git filter-branch` alternative in jj.

**Comment by ilyagr** [comment]
It's not implemented yet, but the general intention is for [`jj run`](https://martinvonz.github.io/jj/prerelease/design/run/) to do it.

I've also tried to use `git branchless test` for this, with mixed results: https://github.com/arxanas/git-branchless/issues/1406#issuecomment-2381075535. I wouldn't recommend this approach unless the automation saves you a lot of time, e.g. if there are truly many commits to process, it's a bit finicky in a co-located `jj` repo, but it would work if it was really needed (and that bug I linked does not apply, can be worked around, or somebody takes the time to fix it).

**Reply by arxanas** [reply]
As a workaround for present, you can also use `git rebase -i`/`git rebase -x` for similar workflows.

**Reply by ilyagr** [reply]
Fascinating! I never knew plain Git has [`git rebase -x`](https://git-scm.com/docs/git-rebase#Documentation/git-rebase.txt--xltcmdgt). Unfortunately, this means I have to re-remember the syntax of the `git rebase` command (to get it to rebase the commits I want to the same place they were, but with `-x`), but it might be worth it for this.

**Comment by ilyagr** [comment]
See also a very similar discussion in https://github.com/martinvonz/jj/discussions/4584 (which I just closed in favor of this one).
